### PR TITLE
Split cdbdisp.c into several files, and put them into a new dispatcher/ directory

### PR DESCRIPTION
**Pull request by eurekaka**
This commit has no logic change, it just contains movement of code across
files, to make dispatcher code clearer, and easier for unit testing.

Signed-off-by: Kenan Yao

### CODE REVIEW

**Comment by andreasscherbaum** [comment]
With Pulse being happy about this PR, I'm +1

File: src/include/cdb/cdbdisp_thread.h [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+ * cdbdisp_thread.h
+ * routines for multi-thread implementation of dispatching commands 
+ * to the qExec processes.
+ *
+ * Copyright (c) 2005-2008, Greenplum inc
+ *
+ *-------------------------------------------------------------------------
+ */
+#ifndef THREADCDBDISP_H
+#define THREADCDBDISP_H
```
Comment (danielgustafsson) [inline src/include/cdb/cdbdisp_thread.h]: This headerguard isn't following the usual namingconvention.

**Comment by danielgustafsson** [comment]
With the headerguard mentioned inline fixed; looks good to me. `cdbdisp.c` is a bit of a monster today so it would be good to get it down to more bite size chunks.

**Comment by xiong-gang** [comment]
+1

**Comment by pengzhout** [comment]
LGTM

**Comment by eurekaka** [comment]
Merged with comments fixed, thanks for reviewing.

**Pull request closed without merging**